include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,481 @@
+    friend bool operator==(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return std::tie(lhs.window_right, lhs.window_values_size) == std::tie(rhs.window_right, rhs.window_values_size);
+    }
+
+    //!\brief Compare to the sentinel of the underlying range.
+    friend bool operator!=(window_iterator const & lhs, sentinel_t const & rhs) noexcept
+    {
+        return !(lhs == rhs);
+    }
+
+    //!\brief Compare to the sentinel of the underlying range.
+    friend bool operator!=(sentinel_t const & lhs, window_iterator const & rhs) noexcept
+    {
+        return !(lhs == rhs);
+    }
+
+    //!\brief Compare to another window_iterator.
+    friend bool operator!=(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return !(lhs == rhs);
+    }
+
+    //!\brief Compare to another window_iterator.
+    friend bool operator<(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return (lhs.window_right < rhs.window_right) && (lhs.window_values_size < rhs.window_values_size);
+    }
+
+    //!\brief Compare to another window_iterator.
+    friend bool operator>(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return (lhs.window_right > rhs.window_right) && (lhs.window_values_size > rhs.window_values_size);
+    }
+
+    //!\brief Compare to another window_iterator.
+    friend bool operator<=(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return (lhs.window_right <= rhs.window_right) && (lhs.window_values_size <= rhs.window_values_size);
+    }
+
+    //!\brief Compare to another window_iterator.
+    friend bool operator>=(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return (lhs.window_right >= rhs.window_right) && (lhs.window_values_size >= rhs.window_values_size);
+    }
+    //!\}
+
+    //!\brief Pre-increment.
+    window_iterator & operator++() noexcept
+    {
+        get_minimiser();
+        return *this;
+    }
+
+    //!\brief Post-increment.
+    window_iterator operator++(int) noexcept
+    {
+        window_iterator tmp{*this};
+        get_minimiser();
+        return tmp;
+    }
+
+    //!\brief Return the minimiser.
+    value_type operator*() const noexcept
+    {
+        return minimiser_value;
+    }
+
+private:
+    //!brief Iterator to last element in range.
+    sentinel_t urange_end;
+
+    //!\brief The minimiser value.
+    size_t minimiser_value{0};
+
+    //!\brief Iterator to the rightmost value of one window.
+    it_t window_right;
+
+    //!\brief The number of values in one window.
+    uint32_t window_values_size;
+
+    //!\brief Stored values per window. It is necessary to store them, because a shift can remove the current minimiser.
+    std::deque<uint64_t> window_values;
+
+    //!\brief Increments iterator by 1.
+    void get_minimiser()
+    {
+        if (window_values.size() == 0)
```

Comment:
If clauses are very expensive in "hot-loops". I'm not sure, but this if-statement is only true at the construction of the window_iterator, right? Then Maybe this if clause can be removed then if you call `window_first` directly in the constructor. 
EDIT:
And on second thought, can you check if we need to store `window_value_size` at all then? When I grep for it in the iterator it is only used in the constructor, `window_first` and the comparison operators. 
In the comparator operators you can use `window_values.size()` directly

Reply:
When I put window_first in the constructor, the test for the forward_range fails, but everything else passes. Are we okay with losing the foward_range property?

Reply:
Hmm this is weird. Can you check why we loose that property? We only change something internally, it should not have an influence on the public API

Reply:
The reason is, that it then uses the input_iterator_tag instead of the forward one. Removing the lines about the iterator_concept solved this issue, so I would just do that?

Reply:
No, I don't think we should remove the iterator category. I don't remember the exact details (@rrahn can you help) but I think this is needed to be correctly identifier as the respective range in c++20.
We need to check why this happens. If you push the latest changes and are done with the rest I take a look if you like.

Reply:
That would be nice, thanks. :) 
I have pushed everything else now.